roundrobin/rr.go

```diff
@@ -62,14 +71,35 @@ func (r *RoundRobin) Next() http.Handler {
 }
 
 func (r *RoundRobin) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	url, err := r.NextServer()
-	if err != nil {
-		r.errHandler.ServeHTTP(w, req, err)
-		return
-	}
 	// make shallow copy of request before chaning anything to avoid side effects
 	newReq := *req
-	newReq.URL = url
+	stuck := false
```

Comment:
Glad I saw your comment in my other PR, I didn't see this comment!

I've retraced the logic of that section.  The flow is this:

* Look to see if we have a sticky session
** if so, pull the server out of the cookie and store it.
** if not, fall through to the next session

* Look to see if we use sticky sessions & do not have a server || if we do not use sticky sessions:
** call NextServer() to get the approp. URL
** if we have sticky sessions, stick that backend

* Return the calculated backend URL

So we only call NextServer() if we have failed to retrieve a server URL from the cookie - that's in the `if !stuck {` stanza.  We won't call it if `stuck = true`, which is only if we have retrieved the server from the cookie -- and, validated that it is still a valid server, which we handle in `ss.GetBackend`.